mlos_bench/mlos_bench/tunables/tunable_groups.py

```diff
@@ -110,10 +110,10 @@ def merge(self, tunables: "TunableGroups") -> "TunableGroups":
                 self._add_group(group)
             else:
                 # Check that there's no overlap in the tunables.
-                if self._tunable_groups[group.name] != group:
+                # But allow for differing current values.
+                if not self._tunable_groups[group.name].equals_defaults(group):
                     raise ValueError(f"Overlapping covariant tunable group name {group.name} " +
```

Comment:
Fixes for a bug mentioned in the conversations about the order of applying tunable_values vs loading and merging more tunable_params from different environments/optimizers/cli/etc. configs.
May need more tests ...